### ISSUE TITLE
Difference between serial and parallel scf energy

### ISSUE THREAD

**Issue by fdmalone** [pyscf/mpi4pyscf#15]
Hi,

I noticed a difference in the RHF energy when running with FFTDF in parallel. The difference is quite small (~0.6 mHa / Cell) but it only appears to happen for larger, or potentially just the gth-cc-XZVP basis sets for X > D. I've pasted the scf script below and attached the output files for both serial and parallel (srun -n 8) calculations.

I'm using mpi4pyscf master and pyscf v1.6.6. The difference only seems to appear for gth-cc-TZVP and gth-cc-QZVP, gth-cc-DZVP (which I think is the same as gth-DZVP) seems fine.

If it's any help the initial energy seems slightly different too.  

```python

#!/usr/bin/env python

import numpy
from functools import reduce
from pyscf.pbc import gto, scf
from pyscf import scf as mol_scf
from mpi4pyscf.pbc import df as mpidf

alat0 = 3.6

cell = gto.Cell()
cell.a = (numpy.ones((3,3))-numpy.eye(3))*alat0/2.0
cell.atom = (('C',0,0,0),('C',numpy.array([0.25,0.25,0.25])*alat0))
cell.basis = 'gth-cctzvp'
cell.pseudo = 'gth-pade'
cell.mesh = [33]*3
cell.rcut = 38
cell.verbose = 5
cell.build()

nk = [2,2,2]
kpts = cell.make_kpts(nk)

cutoff = 1e-14
mf = scf.KRHF(cell,kpts)
# mf = mol_scf.addons.remove_linear_dep_(mf,threshold=cutoff,lindep=cutoff)
mf.with_df = mpidf.FFTDF(cell,kpts)
mf.chkfile = 'scf.dump'
mf.conv_tol = 1e-9
ehf = mf.kernel()

```

**Comment by fdmalone** [comment]
[mpi.txt](https://github.com/pyscf/mpi4pyscf/files/4327786/mpi.txt)
[serial.txt](https://github.com/pyscf/mpi4pyscf/files/4327787/serial.txt)

**Comment by sunqm** [comment]
Fixed in commit 5f16c308db1b899334c47b8fe517186b38f3eec2
